src/main/kotlin/org/rust/cargo/runconfig/test/CargoTestRunConfigurationProducer.kt

```diff
@@ -96,10 +96,15 @@ class TestConfig(
     val configurationName: String,
     testPath: String,
     target: CargoWorkspace.Target,
-    private val exact: Boolean
+    exact: Boolean
 ) {
-    val cargoCommandLine: CargoCommandLine = CargoCommandLine.forTarget(target, "test", listOf(testPath))
-        .let { if (exact) it.withDoubleDashFlag("--exact") else it }
+    val cargoCommandLine: CargoCommandLine = if (exact) {
+        CargoCommandLine.forTarget(target, "test")
+            .withDoubleDashFlag("--exact")
+            .withDoubleDashFlag(testPath)
```

Comment:
I'm not sure if it's supposed to work but it works in practice

Reply:
Hm, I think the bug is fixed in the wrong place in this instance. When running `cargo test foo -- bar`, Cargo runs the test binary as `./target/debug/test-hash foo bar`. That is, the last positional argument and all args after `--`  are passed to the binary. 

I think the logic here needs to be adjusted: https://github.com/intellij-rust/intellij-rust/blob/2bbb484903b726e74e30e5a931bd3f2ba596f995/debugger/src/main/kotlin/org/rust/debugger/runconfig/RsDebugRunner.kt#L63. It should not be `splitOnDoubleDash`, but `getTestBinaryArguments`, and should basically replicate Cargo's argument parsing.